main.go

```diff
@@ -130,93 +121,36 @@ func main() {
 
 	cacherClient, err := client.New(*facility)
 	if err != nil {
-		panic(err)
+		logger.Fatal(err, "Failed to create the cacher client")
 	}
-	go func() {
-		c := time.Tick(15 * time.Second)
-		for range c {
-			// Get All hardware as a proxy for a healthcheck
-			// TODO (patrickdevivo) until Cacher gets a proper healthcheck RPC
-			// a la https://github.com/grpc/grpc/blob/master/doc/health-checking.md
-			// this will have to do.
-			// Note that we don't do anything with the stream (we don't read from it)
-			var isCacherAvailableTemp bool
-			ctx, cancel := context.WithCancel(context.Background())
-			_, err := cacherClient.All(ctx, &cacher.Empty{})
-			if err == nil {
-				isCacherAvailableTemp = true
-			}
-			cancel()
-
-			isCacherAvailableMu.Lock()
-			isCacherAvailable = isCacherAvailableTemp
-			isCacherAvailableMu.Unlock()
-
-			logger.With("status", isCacherAvailableTemp).Debug("tick")
-		}
-
-	}()
 
 	grpcServer := grpc.NewServer(serverOpts...)
-	hegel.RegisterHegelServer(grpcServer, &server{
+	hegelServer = &server{
 		log:          logger,
 		cacherClient: cacherClient,
-	})
+	}
 
-	// Register grpc prometheus server
+	hegel.RegisterHegelServer(grpcServer, hegelServer)
 	grpc_prometheus.Register(grpcServer)
-	http.Handle("/metrics", promhttp.Handler())
-	http.HandleFunc("/_packet/healthcheck", healthCheckHandler)
-	http.HandleFunc("/_packet/version", versionHandler)
-	logger.With("port", *metricsPort).Info("Starting http server")
-	go func() {
-		err := http.ListenAndServe(fmt.Sprintf(":%d", *metricsPort), nil)
-		if err != nil {
-			logger.Error(err, "failed to serve http")
-			panic(err)
-		}
-	}()
 
-	logger.With("listen_address", grpcAddr).Info("starting gRPC server")
-	err = grpcServer.Serve(lis)
+	logger.Info("serving grpc")
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
 	if err != nil {
-		logger.Error(err, "failed to serve grpc")
+		err = errors.Wrap(err, "failed to listen")
+		logger.Error(err)
 		panic(err)
 	}
-}
-
-func versionHandler(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json")
-	//nolint:errcheck
-	w.Write(gitRevJSON)
-}
 
-func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
-	isCacherAvailableMu.RLock()
-	isCacherAvailableTemp := isCacherAvailable
-	isCacherAvailableMu.RUnlock()
-
-	res := struct {
-		GitRev          string  `json:"git_rev"`
-		Uptime          float64 `json:"uptime"`
-		Goroutines      int     `json:"goroutines"`
-		CacherAvailable bool    `json:"cacher_status"`
-	}{
-		GitRev:          gitRev,
-		Uptime:          time.Since(startTime).Seconds(),
-		Goroutines:      runtime.NumGoroutine(),
-		CacherAvailable: isCacherAvailableTemp,
-	}
-	b, err := json.Marshal(&res)
+	err = grpcServer.Serve(lis)
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
+		logger.Fatal(err, "Failed to serve  grpc")
 	}
 
-	if !isCacherAvailableTemp {
-		w.WriteHeader(http.StatusInternalServerError)
-	}
```

Comment:
If hegel can't connect to the underlying data store (either cacher or tink), this seems like something we would still want to handle.

Reply:
All of these http calls calls and function are there as it was before. They are just moved from main.go to http_handle.go